File: backend/routes/stats.py
"""
统计相关路由
"""

from flask import Blueprint, request, jsonify
from backend.models.questions import QuestionsModel

stats_bp = Blueprint('stats', __name__)


@stats_bp.route('/api/stats', methods=['GET'])
def get_stats():
    """获取统计信息"""
    stats = QuestionsModel.get_stats()
    return jsonify({
        "success": True,
        "stats": stats
    })


@stats_bp.route('/api/chapters', methods=['GET'])
def get_chapters():
    """获取所有章节列表（保持原始顺序）"""
    bank = request.args.get('bank', '')
    chapters = QuestionsModel.get_chapters(bank=bank if bank else None)
    return jsonify({
        "success": True,
        "chapters": chapters
    })


@stats_bp.route('/api/stats/by_bank', methods=['GET'])
def get_stats_by_bank():
    """获取按题库分组的统计信息"""
    stats = QuestionsModel.get_stats_by_bank()
    return jsonify({
        "success": True,
        "stats": stats
    })

File: frontend/js/modules/stats.js
/**
 * 统计数据模块
 */

async function loadStats() {
    try {
        const data = isElectron ?
            await window.electronAPI.getStats() :
            await (await fetch(`${API_BASE}/api/stats`)).json();
        
        if (data.success) {
            const stats = data.stats;
            document.getElementById('total-banks').textContent = stats.total_banks;
            document.getElementById('total-questions').textContent = stats.total_questions;
            document.getElementById('single-count').textContent = stats.single_choice_count;
            document.getElementById('multi-count').textContent = stats.multi_choice_count;
        }
    } catch (error) {
        console.error('加载统计数据失败:', error);
    }
}

async function loadBankChapters() {
    try {
        // 暂时复用 getQuestions 获取统计数据
        const data = isElectron ?
            await window.electronAPI.getQuestions() :
            await (await fetch(`${API_BASE}/api/stats/by_bank`)).json();
        
        const container = document.getElementById('bank-chapters-container');
        if (!container) return;
        
        if (data.success && Object.keys(data.stats).length > 0) {
            container.innerHTML = Object.entries(data.stats).map(([bankName, bankData]) => {
                const chaptersHtml = Object.entries(bankData.chapters).map(([chapterName, count]) => `
                    <div class="chapter-item">
                        <span class="chapter-name" title="${chapterName}">${chapterName}</span>
                        <span class="chapter-count">${count}题</span>
                    </div>
                `).join('');
                
                return `
                    <div class="bank-chapter-group">
                        <div class="bank-title">
                            <i class="fas fa-book"></i>
                            ${bankName}
                            <span class="bank-count">(共${bankData.total}题)</span>
                        </div>
                        <div class="chapter-list">
                            ${chaptersHtml}
                        </div>
                    </div>
                `;
            }).join('');
        } else {
            container.innerHTML = `
                <div class="empty-state">
                    <i class="fas fa-inbox"></i>
                    <p>暂无题库，请先导入题库</p>
                </div>
            `;
        }
    } catch (error) {
        console.error('加载题库章节失败:', error);
    }
}
